### PR TITLE
changes to error handling in sms forms

### PR DESCRIPTION
**Pull request by snopoke**
First 2 commits are cleanup.

Goes along with https://github.com/dimagi/formplayer/pull/672

### CODE REVIEW

File: corehq/apps/formplayer_api/smsforms/api.py [hunk 1]
```diff
@@ -373,11 +360,10 @@ def get_raw_instance(session_id, domain=None, auth=None):
         }
 
     response = post_data(json.dumps(data), auth)
-    response = json.loads(response)
     if "error" in response:
         error = response["error"]
-        if error == "invalid session id":
```
Comment (snopoke) [inline corehq/apps/formplayer_api/smsforms/api.py]: Based on where this was being returned from formplayer I don't think it was correct in the first place.

File: corehq/apps/formplayer_api/smsforms/api.py [hunk 2]
```diff
@@ -278,22 +279,8 @@ def server_down(cls):
                                         "contact your administrator for help."})
 
 
-def post_data_helper(d, auth, content_type, url, log=False):
```
Comment (calellowitz) [inline corehq/apps/formplayer_api/smsforms/api.py]: was this just not being used?

File: corehq/apps/formplayer_api/smsforms/api.py [hunk 3]
```diff
@@ -278,22 +278,8 @@ def server_down(cls):
...
-    conn = six.moves.http_client.HTTPConnection(up.netloc)
-    conn.request('POST', up.path, data, headers)
-    resp = conn.getresponse()
-    results = resp.read()
-    return results
-
-
 def formplayer_post_data_helper(d, content_type, url):
     data = json.dumps(d).encode('utf-8')
-    up = urlparse(url)
```
Comment (calellowitz) [inline corehq/apps/formplayer_api/smsforms/api.py]: this too?

**Review by calellowitz — APPROVED** [review]

File: corehq/apps/formplayer_api/smsforms/api.py (hunk 2 above)
Comment (snopoke) [reply corehq/apps/formplayer_api/smsforms/api.py]: yes

File: corehq/apps/formplayer_api/smsforms/api.py (hunk 3 above)
Comment (snopoke) [reply corehq/apps/formplayer_api/smsforms/api.py]: yes

**Pull request merged**